### DISCUSSION TITLE
New API reference site

### DISCUSSION THREAD

**Discussion by louthy** [louthy/language-ext#929] [Announcements]
Last night I knocked up a new API documentation site, as the last one was a bit shonky and so the API docs had fallen well behind the source.  Anyway, it's live now and can be accessed here: [lang-ext.org](http://lang-ext.org) or here: [louthy.github.io/language-ext](https://louthy.github.io/language-ext)

It is still generated from the code, and doesn't have exceptional documentation throughout.  I'll work on that where I can.  It's also v0.1 of the docs site (literally built it in a few hours last night); I'll try and get some more advanced features in there if I can (at least a search of some sort!)

I've also updated all the links from the README.md, so they should all point to the correct API documentation.

Here's a small example:

![image](https://user-images.githubusercontent.com/1991827/137955475-b7e0288b-cf30-4d2f-8f2e-671da98573c9.png)

**Comment by MrYossu** [comment]
@louthy I just love the word "shonky"!!

Thanks as ever for all your efforts. The documentation will help overcome a huge barrier for those of us still struggling to learn.

**Comment by MrYossu** [comment]
@louthy Just been browsing the new docs, and have two small requests.

1) Don't know how easy it would be, but it would be really useful to have a list of contents at the top of each page. For example, the page for [LanguageExt.Core ► Prelude](https://louthy.github.io/language-ext/LanguageExt.Core/Prelude/index.html) is pretty long, and it's hard to see what's in there. A clickable list of contents would be very useful.

2) Along similar lines, it would be useful if you could make the sections collapsible. Again, this would help find what you want.

Thanks again, keep up the great work!

**Reply by louthy** [reply]
1. Is on my list, I’ll probably get that done this evening 
2. Probably not, I’d prefer to make navigation easier

**Comment by louthy** [comment]
Updated:

* A 'Contents' links section on each page
* Added anchor links to each declaration so they can be copied and shared
* Re-arranged more of the source code so it's easier to get around the documentation.  
    * Especially the `Prelude` which was massive, it's now been broken up into sub-sections; same with Pipes which was pretty massive too.

![image](https://user-images.githubusercontent.com/1991827/138001230-1081e79a-c155-48a6-b0a5-2721a0b20f83.png)

**Comment by louthy** [comment]
Another update to the docs this evening.  I updated my doc-building tool to look for `readme.md` in the folders, if it finds one it uses it as a preamble on the page, which allows me to set the context for each sub-folder's worth of functionality.  That's pretty sweet I think

![image](https://user-images.githubusercontent.com/1991827/138190715-b7e11367-7c2a-4216-95db-40e1631e1d70.png)

![image](https://user-images.githubusercontent.com/1991827/138192246-072ef8f1-bcc0-4b30-af4f-84c201fd299a.png)

![image](https://user-images.githubusercontent.com/1991827/138192308-272be278-8e79-4c9d-bde7-4a103a3fe657.png)

![image](https://user-images.githubusercontent.com/1991827/138192385-553097e4-804b-44d0-a1a9-a3b032fd8366.png)

![image](https://user-images.githubusercontent.com/1991827/138192427-cc420bc1-0823-4756-9f47-d1711d9cc22c.png)

**Comment by MrYossu** [comment]
@louthy That's amazing! This is what we've been needing for ages. Thank you so much for doing this. Please don't stop, we want more and more and more!
